client/landing/gutenboarding/hooks/use-step-navigation.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { Step, usePath, useCurrentStep } from '../path';
+import { STORE_KEY as ONBOARD_STORE } from '../stores/onboard';
+
+/**
+ * A React hook that returns previous and next steps in Gutenboarding flow
+ *
+ * @typedef { object } NavigationPath
+ * @property { string } previousStepPath of the previous step
+ * @property { string } nextStepPath of the next step
+ *
+ * @returns { NavigationPath } An object with paths for previous and next steps
+ */
+export default function useStepNavigation(): { previousStepPath: string; nextStepPath: string } {
+	const makePath = usePath();
+	const currentStep = useCurrentStep();
+	const siteTitle = useSelect( ( select ) => select( ONBOARD_STORE ).getSelectedSiteTitle() );
+
+	const steps = siteTitle
+		? [ Step.IntentGathering, Step.Domains, Step.DesignSelection, Step.Style, Step.Plans ]
+		: [ Step.IntentGathering, Step.DesignSelection, Step.Style, Step.Domains, Step.Plans ];
```

Comment:
Could you please add a comment explaining the business logic behind this? Very well implemented btw.